### PR TITLE
[COMP] Refactored LookupElements.kt

### PR DESCRIPTION
**Pull request by mhaessig**
Performed proposed refactoring from PR #1901 and thus finally resolving issue #1888.

### CODE REVIEW

File: src/main/kotlin/org/rust/lang/core/completion/LookupElements.kt [hunk 1]
```diff
@@ -22,113 +22,131 @@ const val KEYWORD_PRIORITY = 10.0
...
         } else {
             base
         }
 
         is RsConstant -> base
-            .withTypeText(element.typeReference?.text)
-            .withInsertHandler({ context: InsertionContext, _ -> appendSemicolon(context) })
-        is RsFieldDecl -> base.withTypeText(element.typeReference?.text)
-        is RsTraitItem -> base.withInsertHandler({ context: InsertionContext, _ -> appendSemicolon(context) })
+            .withTypeText(this.typeReference?.text)
```
Comment (Undin) [inline src/main/kotlin/org/rust/lang/core/completion/LookupElements.kt]: Explicit `this` here and below can be dropped

File: src/main/kotlin/org/rust/lang/core/completion/LookupElements.kt [hunk 2]
```diff
@@ -22,113 +22,131 @@ const val KEYWORD_PRIORITY = 10.0
 private const val MACRO_PRIORITY = -0.1
 
 fun createLookupElement(element: RsElement, scopeName: String): LookupElement {
-    val base = LookupElementBuilder.create(element, scopeName)
-        .withIcon(if (element is RsFile) RsIcons.MODULE else element.getIcon(0))
+    val base = element.getLookupElementBuilder(scopeName)
+        .withInsertHandler({ context: InsertionContext, _ ->
+            run {
```
Comment (Undin) [inline src/main/kotlin/org/rust/lang/core/completion/LookupElements.kt]: It seems that if you drop explicit `()` at `withInsertHandler` call and `run` function, code would be a little bit clearer

Comment (Undin) [reply src/main/kotlin/org/rust/lang/core/completion/LookupElements.kt]: And maybe extract this lambda as separate function?

File: src/main/kotlin/org/rust/lang/core/completion/LookupElements.kt [hunk 3]
```diff
@@ -22,113 +22,131 @@ const val KEYWORD_PRIORITY = 10.0
...
+                        context.document.insertString(context.selectionEndOffset, "!$parens")
+                        EditorModificationUtil.moveCaretRelatively(context.editor, 2)
+                    }
+
+                }
+            }
+        })
+
+    if (element is RsMacroDefinition) {
+        return base.withPriority(MACRO_PRIORITY)
```
Comment (Undin) [inline src/main/kotlin/org/rust/lang/core/completion/LookupElements.kt]: We usually use `if (condition) return value` pattern in such cases

**Comment by mhaessig** [comment]
Incorporated changes suggested in the review.

**Comment by Undin** [comment]
bors r+
Thanks!

**Pull request merged**